app/main.py

```diff
@@ -2,0 +1,26 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name, health=100, hidden=False):
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def die(self):
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self):
+        return f"{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}"
+
+class Herbivore(Animal):
+    def hide(self):
+        self.hidden = not self.hidden
+
+class Carnivore(Animal):
+    def bite(self, animal):
+        if isinstance(animal, Herbivore) and not animal.hidden:
```

Comment:
The `bite` method correctly checks if the `animal` is an instance of `Herbivore` and not hidden before reducing its health. This logic is appropriate for the intended functionality.